module/modmanager/manager_test.go

```diff
@@ -360,6 +362,49 @@ func TestModManagerFunctions(t *testing.T) {
 	}
 }
 
+func TestModManagerKill(t *testing.T) {
+	// this test will not pass on windows as it relies on the UnixPid of the managed process
+	if runtime.GOOS == "windows" {
+		t.Skip()
+	}
+	modPath := rtestutils.BuildTempModule(t, "examples/customresources/demos/simplemodule")
+	logger, logs := logging.NewObservedTestLogger(t)
+	parentAddr := setupSocketWithRobot(t)
+
+	ctx := context.Background()
+	mgr := setupModManager(t, ctx, parentAddr, logger, modmanageroptions.Options{})
+	modCfg := config.Module{
+		Name:    "simple-module",
+		ExePath: modPath,
+	}
+	err := mgr.Add(ctx, modCfg)
+	test.That(t, err, test.ShouldBeNil)
+
+	// get the module from the module map
+	mMgr, ok := mgr.(*Manager)
+	test.That(t, ok, test.ShouldBeTrue)
+
+	mod, ok := mMgr.modules.Load(modCfg.Name)
+	test.That(t, ok, test.ShouldBeTrue)
+
+	mgr.Kill()
+
+	testutils.WaitForAssertion(t, func(tb testing.TB) {
+		test.That(tb, logs.FilterMessageSnippet("Killing module").Len(),
```

Comment:
not the most robust test, but using the Status() method didn't work

Reply:
> Using the `Status()` method didn't work.

Any idea why? Still reported being alive?

Reply:
err is returning nil, so I assume so. Maybe a quirk of how it works on CI, because the test with Status() worked fine locally